tasks/bundle_analysis_processor.py

```diff
@@ -106,20 +107,29 @@ def process_impl_within_lock(
         )
         assert commit, "commit not found"
 
+        report_service = BundleAnalysisReportService(commit_yaml)
+
+        # these are the task results from prior processor tasks in the chain
+        # (they get accumulated as we execute each task in succession)
+        processing_results = previous_result.get("results", [])
+
         # these are populated in the upload task
+        # unless when this task is called on a non-BA upload then we have to create an empty upload
         upload_pk = params["upload_pk"]
-        assert upload_pk is not None
+        if upload_pk is None:
+            commit_report = async_to_sync(report_service.initialize_and_save_report)(
+                commit
+            )
+            upload_pk = report_service.create_report_upload(
```

Comment:
It seems these can have a state of `uploaded` or `carryforward`. I think this is one that was carried forward?

Reply:
Updated. The upload_type[_id] fields are now carriedforward in this case, implemented in the `update_upload` function. (state field is for started/processed flags).